doc/melange.rst

```diff
@@ -4,49 +4,188 @@
 JavaScript Compilation With Melange
 ***********************************
 
+Introduction
+------------
+
 `Melange <https://github.com/melange-re/melange>`_ is a compiler from OCaml to
 JavaScript. Unlike js_of_ocaml, Melange works by translating OCaml compiler
 internal lambda representation to JS files. This allows to produce a single
 JavaScript file from each OCaml module. Melange can be installed with
-`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_ package
-managers.
+`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_
+package managers.
+
+Dune can build projects using Melange, and it allows to generate JavaScript
+files by defining a :ref:`melange-emit` stanza. Dune libraries can be used
+with Melange, by leveraging the ``(modes melange)`` field in the :ref:`library`
+stanza.
+
+To enable Melange support in a Dune project, specify the Melange language
+version in the :ref:`dune-project` file. Adding
+
+.. code:: scheme
+
+    (using melange 0.1)
+
+to a :ref:`dune-project` file enables using the ``melange.emit`` stanza and the
+``(modes melange)`` field for the :ref:`library` stanza.
+
+.. _melange-emit:
+
+melange.emit
+------------
+
+The Melange emit stanza allows to generate JavaScript from Melange libraries and
+modules. It could be seen as the OCaml :ref:`executable` stanza, but for Melange
+projects:
+
+.. code:: scheme
+
+    (melange.emit
+     (target <target>)
+     (module_system <module_system>)
+     <optional-fields>)
 
-Compiling to JS
-===============
+``<target>`` is the name of the target folder that will be used to place or the
+JavaScript artifacts resulting from the build. In particular, the folder will be
+placed under ``_build/default/$path-to-directory-of-melange-emit-stanza``.
 
-Dune has experimental support for building Melange libraries. To use it, you
-must declare the ``melange`` extension in your ``dune-project`` file:
+The JavaScript artifacts will be placed following the file layout of the source
+tree. For example, given the following source tree:
+
+.. code:: bash
+
+    ├── dune # (melange.emit (target output) (libraries lib) (module_system es6))
+    ├── app.ml
+    └── lib
+        ├── dune # (library (name lib) (modes melange))
+        └── helper.ml
+
+The resulting layout in ``_build/default`` will be as follows:
+
+.. code:: bash
+
+    output
+    ├── melange__App.js
+    └── lib
+        ├── lib__.js
+        ├── lib__Helper.js
+        └── lib.js
+
+``<module_system>`` specifies the JavaScript import and export formats used. The
```

Comment:
`format{s,}` because we only allow one format per `melange.emit` stanza?

```suggestion
``<module_system>`` specifies the JavaScript import and export format used. The
```